core/src/prediction/QuantilePredictionStrategy.cpp

```diff
@@ -38,6 +38,10 @@ std::vector<double> QuantilePredictionStrategy::predict(
     const std::unordered_map<size_t, double>& weights_by_sample,
     const Data& train_data,
     const Data& data) const {
+  if (weights_by_sample.empty()) {
```

Comment:
We could pull this up into `DefaultPredictionCollector` so that all prediction strategies get the benefit of returning early. This is what we do [in `OptimizedPredictionCollector`]( https://github.com/grf-labs/grf/blob/master/core/src/prediction/collector/OptimizedPredictionCollector.cpp#L120-L124).

Reply:
Sorry, that makes much more sense, corrected now.